SPHINXsys/src/for_3D_build/high_level_simulation_class/solid_structural_simulation_class.cpp

```diff
@@ -7,8 +7,9 @@
 BodyPartByParticleTriMesh::BodyPartByParticleTriMesh(SPHBody* body, string body_part_name, TriangleMeshShape* triangle_mesh_shape)
 : BodyPartByParticle(body, body_part_name)
 {	
-	body_part_shape_ = new ComplexShape(body_part_name);
-	body_part_shape_->addTriangleMeshShape(triangle_mesh_shape, ShapeBooleanOps::add);
+	ComplexShapeTriangleMesh *mesh = new ComplexShapeTriangleMesh();
+	body_part_shape_ = new ComplexShape(mesh);
```

Comment:
where body_part_shape_ is deleted? please use unique ptr to avoid memory leak

Reply:
Memory management is an critical issue of concern. But by changing body_part_shape to std::unique_ptr will affect a large scope of the code, risking the robustness. I think do it as a separate task will be a choice. @chuvirtonomy